src/toil/wdl/wdltoil.py

```diff
@@ -1248,51 +1309,168 @@ def _call_eager(
         # Return the result as a WDL float value
         return WDL.Value.Float(total_size)
 
-STANDARD_SCHEMES = ["http:", "https:", "s3:", "gs:", "ftp:"]
-REMOTE_SCHEMES = STANDARD_SCHEMES + [TOIL_URI_SCHEME]
-ALL_SCHEMES = REMOTE_SCHEMES + ["file:"]
 
-def is_toil_url(filename: str) -> bool:
-    return is_url_with_scheme(filename, [TOIL_URI_SCHEME])
+def extract_workflow_inputs(environment: WDLBindings) -> List[str]:
+    filenames = list()
 
-def is_file_url(filename: str) -> bool:
-    return is_url_with_scheme(filename, ["file:"])
+    def add_filename(file: WDL.Value.File) -> WDL.Value.File:
+        filenames.append(file.value)
+        return file
 
-def is_standard_url(filename: str) -> bool:
-    return is_url_with_scheme(filename, STANDARD_SCHEMES)
+    map_over_files_in_bindings(environment, add_filename)
+    return filenames
 
 
-def is_remote_url(filename: str) -> bool:
-    """
-    Decide if a filename is a known, non-file kind of URL
+def get_file_sizes(
+    filenames: List[str],
+    file_source: AbstractJobStore,
+    search_paths: Optional[List[str]] = None,
+    import_remote_files: bool = True,
+    execution_dir: Optional[str] = None,
+) -> Dict[str, FileMetadata]:
     """
-    return is_url_with_scheme(filename, REMOTE_SCHEMES)
+    Resolve relative-URI files in the given environment and them then into absolute normalized URIs. Returns a dictionary of WDL file values to a tuple of the normalized URI,
```

Comment:
I don't think `and them then` makes sense.

We might also want to say something different than "WDL file values"; we mean the string values that would appear in the `value` field of a `WDL.Value.File` object. But you could also think that a `WDL.Value.File` object is *itself* a "WDL file value".